cli/azd/pkg/exec/command_runner.go

```diff
@@ -213,11 +213,15 @@ func redactSensitiveData(msg string) string {
 		"access token": {
 			regexp.MustCompile("\"accessToken\": \".*\""),
 			"\"accessToken\": \"<redacted>\"",
+		},
+		"deployment token": {
+			regexp.MustCompile("--deployment-token .*"),
+			"--deployment-token <redacted>",
 		}}
 
 	for _, redactRule := range regexpRedactRules {
 		regMatchString := redactRule.matchString
-		return regMatchString.ReplaceAllString(msg, redactRule.replaceString)
+		msg = regMatchString.ReplaceAllString(msg, redactRule.replaceString)
```

Comment:
We should write some tests for this logic.

Reply:
Would the `redactSensitiveData` method need to be exported from the module for that to work? I'll admit to having written a grand total of like 10 lines of Go in my life and half of them are this PR, so I'm not sure how best to write tests.

Reply:
I didn't realise that `redactSensitiveData` had existing tests against it. I've expanded the scenarios there for this new logic.